backend/tests/functional/convert_object_type/test_convert_repositories.py

```diff
@@ -0,0 +1,410 @@
+                        "source_field_name": "operational_status",
+                        "relationship_cardinality": None,
+                    },
+                    "sync_status": {
+                        "is_mandatory": False,
+                        "source_field_name": "sync_status",
+                        "relationship_cardinality": None,
+                    },
+                    "credential": {
+                        "is_mandatory": False,
+                        "source_field_name": "credential",
+                        "relationship_cardinality": "one",
+                    },
+                    "tags": {"is_mandatory": False, "source_field_name": "tags", "relationship_cardinality": "many"},
+                    "transformations": {
+                        "is_mandatory": False,
+                        "source_field_name": "transformations",
+                        "relationship_cardinality": "many",
+                    },
+                    "queries": {
+                        "is_mandatory": False,
+                        "source_field_name": "queries",
+                        "relationship_cardinality": "many",
+                    },
+                    "checks": {
+                        "is_mandatory": False,
+                        "source_field_name": "checks",
+                        "relationship_cardinality": "many",
+                    },
+                    "generators": {
+                        "is_mandatory": False,
+                        "source_field_name": "generators",
+                        "relationship_cardinality": "many",
+                    },
+                    "member_of_groups": {
+                        "is_mandatory": False,
+                        "source_field_name": "member_of_groups",
+                        "relationship_cardinality": "many",
+                    },
+                    "subscriber_of_groups": {
+                        "is_mandatory": False,
+                        "source_field_name": "subscriber_of_groups",
+                        "relationship_cardinality": "many",
+                    },
+                }
+            }
+        }
+
+        # Create a read-only repository and convert it to read-write
+
+        with patch("infrahub.git.tasks.lock"):
+            client_repository = await client.create(
+                kind=InfrahubKind.READONLYREPOSITORY,
+                data={"name": "car-dealership", "location": f"{git_repos_source_dir_module_scope}/car-dealership"},
+            )
+            await client_repository.save()
+
+        repository: CoreReadOnlyRepository = await NodeManager.get_one(
+            db=db, id=client_repository.id, kind=InfrahubKind.READONLYREPOSITORY, raise_on_error=True
+        )
+
+        assert repository.commit.value
+        assert repository.internal_status.value == "active"
+        # assert repository.operational_status.value == "online"  --> it's unknown, is this an issue?
```

Comment:
Is this an issue in the tests or something that we'd see in production as well?

Reply:
When we create a ReadOnlyRepository, `fetch` method is not called, which is responsible for setting the operational status, while it's the case for a read write Repository. I believe this may currently happen in production. I am not sure whether `InfrahubReadOnlyRepository.sync_from_remote` should call this fetch method or if this status should be set through another way.